shared/src/main/scala/squants/Dimension.scala

```diff
@@ -55,8 +55,32 @@ trait Dimension[A <: Quantity[A]] {
    * @param value the source string (ie, "10 kW") or tuple (ie, (10, "kW"))
    * @return Try[A]
    */
-  protected def parse(value: Any) = value match {
-    case s: String              => parseString(s)
+  protected def parse(value: Any): Try[A] = value match {
```

Comment:
Scala Native can't handle regexes with unicode yet, so I added an alternate parsing method for native only.